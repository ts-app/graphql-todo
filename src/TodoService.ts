import { Resolver } from '@ts-app/graphql'

export type Todo = {
  id: string
  title: string
  completed: boolean
}

export class TodoService {
  @Resolver()
  async todos (): Promise<Todo[]> {
    return [
      { id: '1', title: 'abc', completed: true },
      { id: '2', title: 'def', completed: false },
      { id: '3', title: 'ghi', completed: false }
    ]
  }
}
