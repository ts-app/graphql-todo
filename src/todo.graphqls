type Query {
    todos: TodosPayload
}

type Mutation {
    createTodo(title: String!): CreateTodoPayload
    updateTodo(ownerId: String!, id: String!, title: String!): NoPayload
    deleteTodo(ownerId: String!, id: String!): NoPayload
    markCompleted(ownerId: String!, id: String!, completed: Boolean): NoPayload
    deleteCompleted: NoPayload
}

type Todo {
    id: String!
    title: String!
    completed: Boolean
    createdAt: Date
    ownerId: String
    ownerName: String
}

type TodosPayload {
    error: String
    docs: [Todo]
}

type CreateTodoPayload {
    error: String
    id: String
}
