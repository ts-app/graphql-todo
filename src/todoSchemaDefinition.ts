import { mergeTypeDefs, ResolverService } from '@ts-app/graphql'
import { loadFile } from '@ts-app/common'
import { TodoService } from './TodoService'

export const todoSchemaDefinition = () => {
  const resolver = ResolverService.getInstance()

  const todoService = new TodoService()
  resolver.registerService(todoService)

  const typeDefs = mergeTypeDefs([ loadFile(`${__dirname}/todo.graphqls`) ])

  // create resolvers after all services with "@Resolver" are registered
  const resolvers = resolver.makeResolvers()

  return { resolvers, typeDefs }
}
