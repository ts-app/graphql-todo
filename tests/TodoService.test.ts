import { TodoService } from '../src/TodoService'

describe('TodoService', () => {
  test('todos', async () => {
    const todo = new TodoService()
    const message = await todo.todos()
    expect(message).toMatchSnapshot()
  })
})
