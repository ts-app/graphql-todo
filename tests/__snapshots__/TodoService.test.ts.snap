// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoService todos 1`] = `
Array [
  Object {
    "completed": true,
    "id": "1",
    "title": "abc",
  },
  Object {
    "completed": false,
    "id": "2",
    "title": "def",
  },
  Object {
    "completed": false,
    "id": "3",
    "title": "ghi",
  },
]
`;
